### PR TITLE
Update functions-and-arrays.js

### PR DESCRIPTION
**Pull request by RicardVillalba**
is not finished yet, I'm on it (6 and 7 don work). I dit the pull request just in case.

### CODE REVIEW

File: src/functions-and-arrays.js [hunk 1]
```diff
@@ -1,73 +1,431 @@
 // Iteration #1: Find the maximum
 
+var maxOfTwoNumbers = [5, 7];
+
+function findMax(array) {
+  var max = -Infinity;
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:6]: Not really a need to start from -Infinity but ok I guess hehe. Like 0 would have been fine.

File: src/functions-and-arrays.js [hunk 2]
```diff
@@ -1,73 +1,431 @@
 // Iteration #1: Find the maximum
 
+var maxOfTwoNumbers = [5, 7];
+
+function findMax(array) {
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:5]: - The function should have two separate arguments, not an array as an argument.
- The function is named incorrectly and is not logging the result if the second number is highest.

File: src/functions-and-arrays.js [hunk 3]
```diff
@@ -1,73 +1,431 @@
...
+console.log("result :", result);
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+/* 
+I don't fully understand:
+You can use rest parameter (...) syntax 
+allows us to represent an indefinite number of arguments as an array.*/
+function findLongestWord(...strs) {
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:26]: - This function is spreading an array that is getting from a callback. This was not how it was asked in the instructions.

File: src/functions-and-arrays.js [hunk 4]
```diff
@@ -1,73 +1,431 @@
...
+    "pearl",
+    "orchard",
+    "crackpot"
+  )
+);
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+var sumNumbers = numbers.reduce(function(a, b) {
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:46]: This should have been a function as instructed in the repo.

File: src/functions-and-arrays.js [hunk 5]
```diff
@@ -1,73 +1,431 @@
...
+  return a + b;
+}, 0);
+console.log(sumNumbers);
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+var sumNumbers = numbersAvg.reduce(function(a, b) {
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:54]: This should have been a function as instructed in the repo.

File: src/functions-and-arrays.js [hunk 6]
```diff
@@ -1,73 +1,431 @@
...
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
+];
+// I'm not sure about this
+averageWordLength = wordsArr.join("").length / wordsArr.length;
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:73]: variable is not declared. Also, this should have been a function as instructed in the repo.

**Comment by RicardVillalba** [comment]
Okay, I'm seeing that yesterday it was a disaster. I'll try to fix it.

Thanks a lot!

Ricard

File: src/functions-and-arrays.js [hunk 7]
```diff
@@ -1,73 +1,431 @@
...
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience"
+];
+var x = ["machine"];
+function doesWordExist(wordsFind, x) {
```
Comment (tawebbcn) [inline src/functions-and-arrays.js:115]: function is not returning anything.

**Comment by tawebbcn** [comment]
Some of the Iterations show results that match the asked solution, however, neither the instructions in the repo nor the Jasmine testing were followed properly. It is good that you used more advanced methods like .includes, .sort and .reduce. However, the Lab was about using the resources seen in the lectures. Overall there is good use of logic in the code but for next Labs try to follow instructions correctly and ask us for help if you don't understand something.

**Comment by RicardVillalba** [comment]
Okay, next time I'll have a block I'll ask for help!

**Comment by RicardVillalba** [comment]
Okay, sorry about that, next time I'll have a block I'll ask for help!

**Pull request closed without merging**